diagnostic_rt_qpcr/operation_types/run_qpcr/protocol.rb

```diff
@@ -55,37 +59,79 @@ def main
 
     operations.retrieve.make
 
-    composition = PCRCompositionFactory.build(
-      program_name: @job_params[:program_name]
-    )
-    program = PCRProgramFactory.build(
-      program_name: @job_params[:program_name],
-      volume: composition.volume
-    )
-    thermocycler = ThermocyclerFactory.build(
-      model: @job_params[:thermocycler_model]
-    )
+    available_thermocyclers = Parameter.where(key: 'thermocycler').map do |thermo|
+      JSON.parse(thermo.value)
+    end
 
-    set_up_program(
-      thermocycler: thermocycler,
-      program: program,
-      composition: composition,
-      qpcr: @job_params[:qpcr]
-    )
+    operations.each_slice(available_thermocyclers.length).each do |ops_list|
```

Comment:
Seems a little odd that this will keep going after all the available thermocyclers are taken. For that matter, they aren't necessarily free for use, they're just in the lab. You should probably have the tech specify which thermocycler they are going to use for each operation (of the ones that match the `thermocycler_model` parameter), and then error the remaining operations after all the available thermocyclers are taken. Can use the krill `select` method for getting this input.